### PR TITLE
Update QuestPanel

### PR DESCRIPTION
**Pull request by DMills94**
- Update filters style
- Extract common components between QuestsFilters and TaskFilters
- Improve table responsiveness
- Junk some HTML clutter

### CODE REVIEW

File: src/components/common/FilterButtons.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        const isSelected = selectedValues.includes(value);
+
+        if (isSelected)
+            dispatch(
+                updateFunc({
+                    field: filterName,
+                    value: selectedValues.filter(val => val !== value),
+                })
+            );
+        else dispatch(updateFunc({ field: filterName, value: [...selectedValues, value] }));
```
Comment (chaiinchomp) [inline src/components/common/FilterButtons.js:23]: fyi I am going to turn on eslint errors for one-liner if/else statements 😄 https://eslint.org/docs/2.0.0/rules/curly

File: src/components/common/FilterButtons.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    const dispatch = useDispatch();
+
+    const toggleValue = value => {
+        const isSelected = selectedValues.includes(value);
+
+        if (isSelected)
+            dispatch(
+                updateFunc({
+                    field: filterName,
+                    value: selectedValues.filter(val => val !== value),
```
Comment (chaiinchomp) [inline src/components/common/FilterButtons.js:20]: lodash has a nice util function for this btw https://lodash.com/docs/4.17.15#without doesn't make much difference here but just to mention it

**Review by chaiinchomp — APPROVED** [review]

**Comment by chaiinchomp** [comment]
just wondering as well - are you disabling the precommit hooks? I noticed a few things from one of your other PRs had code style errors. Was curious if the hooks are broken or if you had disabled them 👀

**Comment by DMills94** [comment]
> just wondering as well - are you disabling the precommit hooks? I noticed a few things from one of your other PRs had code style errors. Was curious if the hooks are broken or if you had disabled them 👀

I am not! Maybe they're not running but i'm not getting errors i'll look next time.
Will resolve the conflict and checking linting.

**Comment by chaiinchomp** [comment]
I think it was only `prettier` errors, eslint seems to be running fine. I have my IDE configured to fix prettier issues on save so that hook never has anything to do for me 😅 could be broken and I wouldn't know

**Pull request merged**